libraries/chain/chain_controller.cpp

```diff
@@ -519,7 +524,7 @@ void chain_controller::check_transaction_authorization(const SignedTransaction&
          }
       }
 
-   if ((_skip_flags & skip_transaction_signatures) == false)
+   if (!allow_unused_signatures && (_skip_flags & skip_transaction_signatures) == false)
```

Comment:
We should never use _skip_flags and/or bit math in IF statements, instead we should define a function that returns true/false and call that function.  It will make the code more readable and also eliminate errors if we need the same check in multiple places.